lib/SILGen/ManagedValue.cpp

```diff
@@ -55,8 +55,6 @@ ManagedValue ManagedValue::forForwardedRValue(SILGenFunction &SGF,
 /// Emit a copy of this value with independent ownership.
 ManagedValue ManagedValue::copy(SILGenFunction &SGF, SILLocation loc) const {
   auto &lowering = SGF.getTypeLowering(getType());
-  if (lowering.isTrivial())
-    return *this;
```

Comment:
This is the critical piece that was different between `copyUnmanaged` vs `copy`. Anywhere `copyUnmanaged` was, would unconditionally copy an address using `copy_addr`, but `copy` itself would not, because address are "trivial" values.

There's at least one test, `SILGen/read_accessor.swift` that is currently failing in FileCheck because of a new, unnecessary `copy_addr` of the `@in_guaranteed` argument for self `%1 : $*TestKeyPath` to invoke the read accessor in a getter thunk.